internal/utils/utils.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"strings"
+
+	"github.com/charmbracelet/lipgloss"
+)
+
+// LipglossTruncate truncates a given line based on its lipgloss width.
+func LipglossTruncate(s string, width int) string {
+	var i int
+	for i = 0; i < len(s) && lipgloss.Width(s[:i]) < width; i++ {
+	} //revive:disable-line:empty-block
+	return s[:i]
+}
```

Comment:
I believe you can use `runewidth.Truncate` here.

Reply:
I had some issues where when using the runewidth.Truncate that it didn't return the correct results because it doesn't calculate the width correctly. 😊 But maybe there is another way of fixing this? 

As I remember it was when adding the formatting to the text that I got a bug. I can try to give a reproducible example? 😇

Also if not mistaken lipgloss actually specifies that you should use lipgloss.Width instead of len(string) and runewidth.